### ISSUE TITLE
Filtering by ClinVar clinsig + CLINSIG Confident + ClinVar hits has a bug

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#4996]
Example to reproduce on a local demo:

 ### On SNV variants page, this filter works:

<img width="561" alt="image" src="https://github.com/user-attachments/assets/1fcd72cc-66da-43ae-8e92-1418c2b0dbc5">

And you get 3 variants which have likely benign clinsig

 ### This filter also works

<img width="541" alt="image" src="https://github.com/user-attachments/assets/aed2034c-5434-4e7c-a2b1-f46885b73506">

You get no variants, since all likely benign vars in the demo instance have `no criteria`

### This filter returns all variants with any ClinVar significance, which is wrong:

<img width="562" alt="image" src="https://github.com/user-attachments/assets/b39544da-00b7-4e66-857a-f0b88f67c6c7">

**Comment by dnil** [comment]
Bug is perhaps a strong word since the interface doesn't clearly specify the priority for the "clinvar hits" checkbox. But it is sure hard to figure out a simple expression describing the design.

All three have use cases: being able to fully select/deselect "confident" annotations seems like a good feature, as does selecting any tag. Perhaps that could be an option for the multi select instead/later? Either a button that just selects all, or an "All" option like for the chromosomes?

**Comment by northwestwitch** [comment]
> All three have use cases: being able to fully select/deselect "confident" annotations seems like a good feature, as does selecting any tag. Perhaps that could be an option for the multi select instead/later? Either a button that just selects all, or an "All" option like for the chromosomes?

I would call it a bug, because if you filter:

- variants with clinsig
- likely benign
- high confidence

You would expect no variants returned, not all the variants with any clinsig?

But I also understand the design, which is: return variants with clinsig no matter what. But I start to dislike this criterion more and more because you don't really get the chance to decide for yourself, like: I want to really make sure no variants are present if I filter by those 3 criteria.

Regarding including "All" in the clinsig multiselect, isn't it the same as not selecting anything at all? I think I would rather keep it as it is (with the ClinVar hits checkbox), to be able to easily exclude those variants that have no ClinVar annotations

**Comment by dnil** [comment]
Right, two pictures here, one as it is, one as one might want to refactor it. 

As it is, I would expect the above to return at least any high significance hits, but I feel a conflict in design between "clinvar hits"and "likely benign". It's kind of obvious the one named "clinsig confidence always returned" is inclusive. The others less clear.

I do agree that designs like "always" are really hard to get into a flexible system. It would make sense to refactor to retain good functionality for the clinical filter, but make the interface option a bit less strained. Meeting now! 😓
